src/zeta/aca_oam_port_manager.cpp

```diff
@@ -99,60 +99,56 @@ void Aca_Oam_Port_Manager::_clear_all_data()
 void Aca_Oam_Port_Manager::create_entry_unsafe(uint port_number)
 {
   ACA_LOG_DEBUG("%s", "Aca_Oam_Port_Manager::create_entry_unsafe ---> Entering\n");
-
-  unordered_set<uint> tunnel_ids_table;
-  _oam_ports_table.emplace(port_number, tunnel_ids_table);
+  
+  _oam_ports_cache.emplace(port_number);
 
   ACA_LOG_DEBUG("%s", "Aca_Oam_Port_Manager::create_entry_unsafe <--- Exiting\n");
 }
 
 // update oam_ports_table and add the OAM punt rule also if this is the first port in the VPC
-void Aca_Oam_Port_Manager::add_vpc(uint port_number, uint tunnel_id)
+void Aca_Oam_Port_Manager::add_port_number(uint port_number)
 {
-  ACA_LOG_DEBUG("%s", "Aca_Oam_Port_Manager::add_vpc ---> Entering\n");
+  ACA_LOG_DEBUG("%s", "Aca_Oam_Port_Manager::add_port_number ---> Entering\n");
   // -----critical section starts-----
-  _oam_ports_table_mutex.lock();
-  if (_oam_ports_table.find(port_number) == _oam_ports_table.end()) {
+  _oam_ports_cache_mutex.lock();
+  if (_oam_ports_cache.find(port_number) == _oam_ports_cache.end()) {
     create_entry_unsafe(port_number);
     _create_oam_ofp(port_number);
   }
-  _oam_ports_table[port_number].emplace(tunnel_id);
+  _oam_ports_cache.emplace(port_number);
```

Comment:
I think we should remove line 118.

If port_number is not found, create_entry_unsafe(port_number) would do _oam_ports_cache.emplace(port_number) inside

If port_number is found, not need to emplace another one into  _oam_ports_cache